src/extension/src/client-modules/generation.ts

```diff
@@ -221,22 +48,49 @@ export class Generation extends WizardServant {
     );
   }
 
-  private trackCosmosConnectionStringReplace(startTime: number): void {
-    //log in telemetry how long it took replacement
-    // Temporary Disable
-    this.Telemetry.trackEventWithDuration(TelemetryEventName.ConnectionStringReplace, startTime, Date.now());
-  }
+  private getGenerationData(data: any): IGenerationData {
+    const generationData: IGenerationData = {
+      backendFramework: data.backendFramework,
+      frontendFramework: data.frontendFramework,
+      backendFrameworkLinuxVersion: data.backendFrameworkLinuxVersion,
+      pages: data.pages,
+      path: data.path,
+      projectName: data.projectName,
+      projectType: data.projectType,
+      services: [],
+    };
 
-  private hasAzureServices(services: IServicesGenerationPayload): boolean {
-    return services.appService !== null || services.cosmosDB !== null;
-  }
+    if (data.services?.appService) {
+      const { internalName, subscription, resourceGroup, location, siteName } = data.services.appService;
+
+      const appService: IAppService = {
+        internalName,
+        category: SERVICE_CATEGORY.AZURE,
+        type: SERVICE_TYPE.APPSERVICE,
+        subscription,
+        resourceGroup,
+        location,
+        serviceName: siteName,
+      };
+      generationData.services?.push(appService);
+    }
+
+    if (data.services?.cosmosDB) {
+      const { internalName, subscription, resourceGroup, location, accountName, api } = data.services.cosmosDB;
+
+      const cosmosDB: ICosmosDB = {
+        internalName,
+        category: SERVICE_CATEGORY.AZURE,
+        type: SERVICE_TYPE.COSMOSDB,
+        subscription,
+        resourceGroup,
+        location,
+        serviceName: accountName,
+        api,
+      };
+      generationData.services?.push(cosmosDB);
+    }
 
-  //TODO: Move to telemetryService?
-  private deployWithTelemetry<T>(telemetryEvent: string, callback: Promise<T>): Promise<any> {
```

Comment:
this function _seems_ duplicated both in GenerationService & ResourceGroupGenerator. Could it make sense make it public and have it just in one place? Not sure where it fits most to be honest... I´ll think about it..